x-pack/plugins/apm/public/context/UrlParamsContext/index.tsx

```diff
@@ -50,6 +52,9 @@ const UrlParamsContext = createContext({
 
 const UrlParamsProvider: React.ComponentClass<{}> = withRouter(
   ({ location, children }) => {
+    const { core } = useApmPluginContext();
+    setTimePickerDefaultValues({ uiSettings: core.uiSettings });
```

Comment:
I'm not sure this is the right approach. I'd really like us to move away from url params rather than tying us more to it. What do you think about retrieving the timepicker values in the date picker selector, and redirect if this is not already set in the url. Alternatively we could do this before the app is bootstrapped.